src/node_crypto.cc

```diff
@@ -1391,6 +1399,188 @@ void SecureContext::GetCertificate(const FunctionCallbackInfo<Value>& args) {
+  CHECK(psk_val->IsString());
+  Local<String> psk_str = psk_val.As<String>();
+
+  Local<String> enc = String::Empty(isolate);
+  StringBytes::InlineDecoder decoder;
+  if (!decoder.Decode(env, psk_str, enc, HEX))
+    return 0;
+
+  char* psk_buf = decoder.out();
+  size_t psk_len = decoder.size();
+
+  CHECK_LE(psk_len, max_psk_len);
+  memcpy(psk, psk_buf, psk_len);
+
+  return psk_len;
+}
+
+unsigned int SecureContext::PskClientCallback(SSL *ssl,
+                                              const char *hint,
+                                              char *identity,
+                                              unsigned int max_identity_len,
+                                              unsigned char *psk,
+                                              unsigned int max_psk_len) {
+  SecureContext* sc = static_cast<SecureContext*>(
+      SSL_CTX_get_app_data(SSL_get_SSL_CTX(ssl)));
+
+  Environment* env = sc->env();
+  Isolate* isolate = env->isolate();
+  HandleScope scope(isolate);
+
+  Local<Value> argv[] = {
+    Null(isolate),
+    Integer::NewFromUnsigned(isolate, max_psk_len),
+    Integer::NewFromUnsigned(isolate, max_identity_len)
+  };
+  if (hint != nullptr) {
+    MaybeLocal<String> hint_str = String::NewFromUtf8(
+      isolate,
+      hint,
+      v8::NewStringType::kNormal,
+      strlen(hint));
+
+    if (hint_str.IsEmpty()) {
+      return 0;
+    }
+
+    argv[0] = hint_str.ToLocalChecked();
+  }
+  Local<Value> value = sc->object()->Get(env->onpskexchange_string());
+
+  MaybeLocal<Value> maybe_ret;
+  if (value->IsFunction()) {
+    Local<Function> func = value.As<Function>();
+    maybe_ret = func->Call(env->context(), sc->object(), arraysize(argv), argv);
+  } else {
+    return 0;
+  }
+
+  if (maybe_ret.IsEmpty()) {
+    return 0;
+  }
+
+  Local<Value> ret = maybe_ret.ToLocalChecked();
+
+  // The result is expected to be an object. If it isn't, then return 0,
+  // indicating the identity wasn't found.
+  if (!ret->IsObject()) {
+    return 0;
+  }
+  Local<Object> obj = ret.As<Object>();
+
+  Local<Value> psk_val = obj->Get(env->context(),
+                                  env->psk_string()).ToLocalChecked();
+  CHECK(psk_val->IsString());
+  Local<String> psk_str = psk_val.As<String>();
+
+  Local<String> enc = String::Empty(isolate);
+  StringBytes::InlineDecoder decoder;
+  if (!decoder.Decode(env, psk_str, enc, HEX))
+    return 0;
+
+  char* psk_buf = decoder.out();
+  size_t psk_len = decoder.size();
+
+  CHECK_LE(psk_len, max_psk_len);
+  memcpy(psk, psk_buf, psk_len);
+
+  Local<Value> identity_val = obj->Get(
+    env->context(),
+    env->identity_string()).ToLocalChecked();
+  CHECK(identity_val->IsString());
+
+  Local<String> identity_str = identity_val.As<String>();
+  size_t identity_len = identity_str->Length();
+  String::Utf8Value identity_buf(isolate, identity_str);
+
+  CHECK_LE(identity_len, max_identity_len);
+  memcpy(identity, *identity_buf, identity_len);
+
+  return psk_len;
+}
```

Comment:
Most comments on the other function apply to this one as well.  It looks like they could share quite a bit of code.